### DISCUSSION TITLE
Cannot convert `Null` to `Config`

### DISCUSSION THREAD

**Discussion by xplatformed** [wezterm/wezterm#6117] [General]
Create lua config file when I reload wezterm terminal i get this error message and default terminal opens error message in its full context any help would be appreciated: 

Error converting lua value returned by script /Users/
myusername/.wezterm.lua to Config struct: error converting Lua nil to
config::config::Config (Cannot convert `Null` to `Config`)

**Comment by bew** [comment]
Did you `return config` at the end of your config file?

Otherwise please give us your config so we can see where the error might be.

**Reply by xplatformed** [reply]
yes

**Comment by xplatformed** [comment]
sorry have to paste code here as its a hidden config file

**Comment by xplatformed** [comment]
-- Pull in the wezterm API
local wezterm = require("wezterm")

-- This will hold the configuration.
local config = wezterm.config_builder()

-- This is where you actually apply your config choices
-- config.color_scheme = "Batman"

config.font = wezterm.font("MesloLGS Nerd Font Mono")
config.font_size = 19

config.enable_tab_bar = false

config.window_decorations = "RESIZE"

config.window_background_opacity = 0.8
config.macos_window_background_blur = 10
my coolnight colorscheme:
config.colors = {
	foreground = "#CBE0F0",
	background = "#011423",
	cursor_bg = "#47FF9C",
	cursor_border = "#47FF9C",
	cursor_fg = "#011423",
	selection_bg = "#033259",
	selection_fg = "#CBE0F0",
	ansi = { "#214969", "#E52E2E", "#44FFB1", "#FFE073", "#0FC5ED", "#a277ff", "#24EAF7", "#24EAF7" },
	brights = { "#214969", "#E52E2E", "#44FFB1", "#FFE073", "#A277FF", "#a277ff", "#24EAF7", "#24EAF7" },
}
-- and finally, return the configuration to wezterm
return config

**Reply by bew** [reply]
This config file works for me on Linux X11 with version `wezterm 20240203-110809-5046fc22`.
(compared to your config, I only commented `my coolnight colorscheme:` & `config.font = ....` lines and indented colors table)
```lua
-- Pull in the wezterm API
local wezterm = require("wezterm")

-- This will hold the configuration.
local config = wezterm.config_builder()

-- This is where you actually apply your config choices
-- config.color_scheme = "Batman"

-- config.font = wezterm.font("MesloLGS Nerd Font Mono")
config.font_size = 19

config.enable_tab_bar = false

config.window_decorations = "RESIZE"

config.window_background_opacity = 0.8
config.macos_window_background_blur = 10
-- my coolnight colorscheme:
config.colors = {
  foreground = "#CBE0F0",
  background = "#011423",
  cursor_bg = "#47FF9C",
  cursor_border = "#47FF9C",
  cursor_fg = "#011423",
  selection_bg = "#033259",
  selection_fg = "#CBE0F0",
  ansi = { "#214969", "#E52E2E", "#44FFB1", "#FFE073", "#0FC5ED", "#a277ff", "#24EAF7", "#24EAF7" },
  brights = { "#214969", "#E52E2E", "#44FFB1", "#FFE073", "#A277FF", "#a277ff", "#24EAF7", "#24EAF7" },
}
-- and finally, return the configuration to wezterm
return config
```

Try to put that config in a file like `example-wezterm.lua` and run wezterm with it using `wezterm --config-file ./example-wezterm.lua`

If that works try to understand what is different from your own config.

**Comment by xplatformed** [comment]
using customized  color scheme but that should not cause this config error

**Comment by neogeographica** [comment]
You need to comment the "my coolnight colorscheme:" line.

**Comment by xplatformed** [comment]
just logged back on yes your example worked i made the change to my config and now it also works. I assume the uncommented colorscheme is what was throwing the error  at compile time  and thanks  for the help .
